discovery/discovery_test.go

```diff
@@ -350,20 +350,106 @@ func TestNewDBOrchestorPoolCache_PollOrchestratorInfo(t *testing.T) {
 		require.Contains(testOrchs, test)
 	}
 
-	cacheRefreshInterval = 1 * time.Second
+	cacheRefreshInterval = 1 * time.Millisecond
 	returnInfo = polledOrchInfo
-	pool.PollOrchestratorInfo(nil)
-	time.Sleep(3 * time.Second)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	pool.PollOrchestratorInfo(ctx, nil)
+	time.Sleep(5 * time.Millisecond)
 	dbOrchs, err = pool.store.SelectOrchs(nil)
 	require.Nil(err)
 	require.Len(dbOrchs, 3)
 	expPrice, _ = common.PriceToFixed(big.NewRat(1, 1))
 	for _, o := range dbOrchs {
-		t.Log(o)
 		assert.Equal(o.PricePerPixel, expPrice)
 	}
 }
 
+func TestNewDBOrchestorPoolCache_PollOrchestratorInfo_CallCount(t *testing.T) {
+	var mu sync.Mutex
+	first := true
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
+		mu.Lock()
+		if first {
+			time.Sleep(100 * time.Millisecond)
+			first = false
+		}
+		mu.Unlock()
+		return &net.OrchestratorInfo{
+			Transcoder: "transcoderFromTest",
+			PriceInfo: &net.PriceInfo{
+				PricePerUnit:  1,
+				PixelsPerUnit: 1,
+			},
+		}, nil
+	}
+
+	dbh, dbraw, err := common.TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	require := require.New(t)
+	assert := assert.New(t)
+	require.Nil(err)
+
+	// adding orchestrators to DB
+	addresses := []string{"https://127.0.0.1:8936", "https://127.0.0.1:8937", "https://127.0.0.1:8938"}
+	orchestrators := StubOrchestrators(addresses)
+	testOrchs := make([]orchTest, 0)
+	for _, o := range orchestrators {
+		to := orchTest{
+			EthereumAddr:  o.Address.String(),
+			ServiceURI:    o.ServiceURI,
+			PricePerPixel: 1000,
+		}
+		testOrchs = append(testOrchs, to)
+	}
+
+	ethClient := &eth.StubClient{Orchestrators: orchestrators}
+
+	pool := NewDBOrchestratorPoolCache(dbh, ethClient, &stubRoundsManager{}, &stubTicketParamsValidator{})
+
+	// Add orchestrators to DB
+	err = pool.CacheTranscoderPool(nil)
+	require.Nil(err)
+
+	// Ensure orchestrators exist in DB
+	require.Equal(pool.Size(), 3)
+	dbOrchs, err := pool.store.SelectOrchs(nil)
+	require.Nil(err)
+	require.Len(dbOrchs, 3)
+	for _, o := range dbOrchs {
+		test := toOrchTest(o.EthereumAddr, o.ServiceURI, o.PricePerPixel)
+		require.Contains(testOrchs, test)
+	}
+
+	callCount := 0
+	serverGetOrchInfo = func(ctx context.Context, bcast common.Broadcaster, orchestratorServer *url.URL) (*net.OrchestratorInfo, error) {
+		mu.Lock()
+		if first {
+			time.Sleep(100 * time.Millisecond)
+			first = false
+		}
+		mu.Unlock()
+		callCount++
+		return &net.OrchestratorInfo{
+			Transcoder: "transcoderFromTest",
+			PriceInfo: &net.PriceInfo{
+				PricePerUnit:  1,
+				PixelsPerUnit: 1,
+			},
+		}, nil
+	}
+
+	cacheRefreshInterval = 200 * time.Millisecond
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	pool.PollOrchestratorInfo(ctx, nil)
+	time.Sleep(1100 * time.Millisecond)
+
+	// called serverGetOrchInfo 3 + 1100 / 200 * 3 = 18 times
```

Comment:
I'm wary of checking for an exact # of polls here as that feels like it could result in fragile test. What about just checking that it was called >= 2 times?

Reply:
I did run the test quite a few times and didn't run across any anomalies but if it's really a concern we can do a   `16<count<20` range or something like that?

Reply:
Ok yeah nvm no goroutines are used here so I don't think any racy behavior should occur. 

Side note: While confirming this I noticed that the tests for the `discovery` package will fail if run consecutive times using `go test -count=100 -failfast` because the package variable `perm` is being set in the tests without re-assigned back to the original implementation. I have a fix for this that I can push up separately though.

Reply:
Seems like you're right, ran into a discrepancy although not very often. I assumed the 100ms extra 'buffer' would have been sufficient to avoid this, appears not. 

Using a range in ecacd7d5 (`16<=count=<20`)